doc/Guide to Contributing to Sia.md

```diff
@@ -0,0 +1,183 @@
+# Contributing to Sia
+
+#### Table of Contents
+* [Get started with Go](#go)
+  * [Install Go](#install-go)
+  * [Learn Go]("learn-go")
+* [Build Sia](#build-sia)
+* [Contribute to the codebase](#contribute)
+  * [Set up git](#setup-git)
+  * [Fork the Sia repository](#fork)
+  * [Write some code](#write)
+  * [Make a pull request](#pull)
+  * [More git resources](#git-resources)
+* [Where to start](#where-to-start)
+* [Contact us](#contact)
+
+<a name="go"/>
+## Get started with Go
+
+<a name="install-go"/>
+### Install Go
+To install Go on your computer, follow the [official installation guide][install-go].  
+
+You should install the latest [official Go binary][binary] for your system (if not available, [install from source][source]).
+If you plan to cross compile Sia, see [Cross Compilation with Go 1.5][cross] by Dave Cheney.  
+
+Now make a workspace directory in which you will store source code and dependencies.  You can choose any filepath except where you installed Go (don't choose `/usr/local`).
+```bash
+# make a working directory called golang in your home directory
+$ mkdir $HOME/golang
+# store base path in an environmental variable
+$ echo 'export GOROOT=$HOME/golang' >> $HOME/.profile
+# add bin subdirectory to PATH environmental variable
+$ echo 'export PATH=$PATH:$GOROOT/bin' >> $HOME/.profile
+```
+Make a base path for your code.  (You'll need [a Github account][signup].)
+```
+$ mkdir -p $GOPATH/src/GitHub.com/<your GitHub username>
+```
+
+<a name="learn-go"/>
+### Learn Go
+* To get familiar with the language, start with the official [Tour of Go][tour].
+* Move onto [How to Write Go Code][how] to learn how to organize Go packages and use the go tool.
+* Finish with the [Effective Go][effective] guide.
+
+<a name="build-sia"/>
+## Build Sia
+To build Sia on your machine, enter the following on the command line:
+```bash
+# Download Sia and its dependencies
+# Binaries will be installed in $GOROOT/bin
+$ go get -u GitHub.com/NebulousLabs/Sia/...
+
+# Switch to directory containing Sia source code
+$ cd $GOROOT/src/GitHub.com/NebulousLabs/Sia
```

Comment:
`github.com` should be lowercase for these commands. Also, `GOPATH`, not `GOROOT`